src/adm/filter.rs

```diff
@@ -107,6 +119,90 @@ impl AdmFilter {
         l_sentry::report(&merged_tags, sentry::event_from_error(error));
     }
 
+    /// check to see if the bucket has been modified since the last time we updated.
+    pub async fn requires_update(&self) -> HandlerResult<bool> {
+        // don't update non-bucket versions (for now)
+        if !self.is_cloud() {
+            return Ok(false);
+        }
+        if let Some(bucket) = &self.source_url {
+            let host = bucket
+                .host()
+                .ok_or_else(|| {
+                    HandlerError::internal(&format!("Missing bucket Host {:?}", self.source))
+                })?
+                .to_string();
+
+            let obj = cloud_storage::Object::read(&host, bucket.path())
+                .await
+                .map_err(|e| {
+                    HandlerError::internal(&format!(
+                        "Could not read bucket {:?}, {:?}",
+                        self.source, e
+                    ))
+                })?;
```

Comment:
Also: `HandlerError`'s have a `From` for CloudStorage errors now:
```suggestion
                .await?;
```